.azure-pipelines/template-commonlib.yml

```diff
@@ -1,40 +1,59 @@
+parameters:
+- name: archs
+  type: object
+  default:
+    - amd64
+    - armhf
+    - arm64
+
 jobs:
-- job: Build
-  timeoutInMinutes: 120
-  pool: sonicbld
-  variables:
-  - template: template-variables.yml
-  steps:
-  - checkout: self
-    clean: true
-    submodules: recursive
-  - script: |
-      set -ex
-      branch=$(Build.SourceBranchName)
-      # DIST_MASTER is set in variable.
-      BRANCH=DIST_${branch^^}
-      bldenvs=${!BRANCH}
-      [ "$bldenvs" == "" ] && bldenvs="$(COMMON_LIB_BUILD_ENVS)"
-      for bldenv in $bldenvs
-      do
-        BLDENV=$bldenv make -f Makefile.work configure PLATFORM=vs ENABLE_DOCKER_BASE_PULL=y
-      done
-      set +x
-      echo "##vso[task.setvariable variable=bldenvs;]$bldenvs"
-    displayName: Make configure
-  - script: |
-      set -ex
-      for bldenv in $(bldenvs)
-      do
-        LIBNL3_VERSION_BASE=$(grep "LIBNL3_VERSION_BASE =" rules/libnl3.mk | awk '{print$3}')
-        LIBNL3_VERSION=$(grep "LIBNL3_VERSION =" rules/libnl3.mk | awk '{print$3}' | sed -e "s/(//" -e "s/)//" -e "s/\\$//" -e "s/LIBNL3_VERSION_BASE/$LIBNL3_VERSION_BASE/")
-        SONIC_BUILD_JOBS=$(nproc) BLDENV=$bldenv make -f Makefile.work target/debs/$bldenv/libnl-3-200_${LIBNL3_VERSION}_amd64.deb ENABLE_DOCKER_BASE_PULL=y
+- ${{ each arch in parameters.archs }}:
+  - job: Build_${{ arch }}
+    timeoutInMinutes: 120
+    ${{ if eq(arch,'amd64') }}:
+      pool: sonicbld
+    ${{ else }}:
+      pool: sonicbld-${{ arch }}
+    variables:
+    - template: template-variables.yml
```

Comment:
Move to outside of the job? looks like all the jobs are using the variable template.